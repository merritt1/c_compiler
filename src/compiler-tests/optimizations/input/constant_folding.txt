void syscall_print_int(int integer);
void syscall_print_string(char *string);
int syscall_read_int(void);
void syscall_read_string(char *buffer, int length);
void syscall_exit(void);

void main(void) {
	int a;
	a = 123+456;
	a = 123-23;
	a = -3;
	a = -4*20;
	a = 1 << 5;
	a = 2 >> 1;
	a = 7 & 3;
	a = 8 | 3;
	a = 1 + 3 + 6 / 2;
}
