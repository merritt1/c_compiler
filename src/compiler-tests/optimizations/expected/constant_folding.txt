(procBegin,  main)
(addressOf,     r0,      a)
(constInt,     r1,        579)
(storeWord,     r0,      r1)
(addressOf,     r2,      a)
(constInt,     r3,        100)
(storeWord,     r2,      r3)
(addressOf,     r4,      a)
(constInt,     r5,         -3)
(storeWord,     r4,      r5)
(addressOf,     r6,      a)
(constInt,     r7,        -80)
(storeWord,     r6,      r7)
(addressOf,     r8,      a)
(constInt,     r9,         32)
(storeWord,     r8,      r9)
(addressOf,     r10,      a)
(constInt,     r11,          1)
(storeWord,     r10,      r11)
(addressOf,     r12,      a)
(constInt,     r13,          3)
(storeWord,     r12,      r13)
(addressOf,     r14,      a)
(constInt,     r15,         11)
(storeWord,     r14,      r15)
(addressOf,     r16,      a)
(constInt,     r17,          7)
(storeWord,     r16,      r17)
(procEnd ,  main)
